tensorpipe/channel/cuda_ipc/context_impl.cc

```diff
@@ -30,6 +31,26 @@ std::string generateDomainDescriptor() {
   auto bootID = getBootID();
   TP_THROW_ASSERT_IF(!bootID) << "Unable to read boot_id";
   oss << bootID.value();
+
+  int deviceCount;
+  TP_CUDA_CHECK(cudaGetDeviceCount(&deviceCount));
+  std::vector<std::string> uuids(deviceCount);
+  for (int i = 0; i < deviceCount; ++i) {
+    cudaDeviceProp props;
+    TP_CUDA_CHECK(cudaGetDeviceProperties(&props, i));
+    std::ostringstream uuid;
+    uuid << std::hex << std::setw(2) << std::setfill('0');
+    for (int j = 0; j < 16; ++j) {
+      uuid << (props.uuid.bytes[j] & 0xff);
```

Comment:
The `& 0xff` is needed because `char` is apparently signed, and `props.uuid.bytes[j]` gets promoted to (signed) `int` with sign-extension.

Reply:
As a general rule, this type of remarks are better put as comments in the code itself, since they will be helpful to future readers too, which won't see this review thread. My philosophy on this is: comments about why the *new* code, in and of itself, is a certain way (i.e., the *outcome* of the change) go into the code; comments about the *change*, that explain the difference between the old and the new version (and thus require context on both) go into the review thread.

Reply:
As for the change itself, I wonder if we can't just reinterpret_cast this (or even static_cast it? not sure) to an unsigned char. Would it be more readable?

Reply:
One more nit (feel free to ignore): nvidia-smi seems to represent the UUID in the "standard" format with sections broken up by dashes (4-2-2-2-6 bytes). It might be nice to match this for easier comparison.